optimum/bettertransformer/transformation.py

```diff
@@ -0,0 +1,192 @@
+# Step 4: If not, yield an error.
+def replace_to_fast(model):
+    r"""
+    Replaces the current model to its `Fast` implementation. Loops recursively into the model and replaces the
+    `Layer` modules with its `Fast` correspondant model
+
+    Args:
+        `model` (`torch.nn.Module`, **required**):
+            The input model to convert
+    Returns:
+        The converted model
+    """
+    for name, module in model.named_children():
+        if len(list(module.children())) > 0:
+            replace_to_fast(module)
+
+        if hasattr(module, "is_decoder"):
+            # Decoders are not supported yet on Better Transformers
+            if module.is_decoder:
+                continue
+
+        class_name = module.__class__.__name__
+        maybe_fast_module = is_module_fast(class_name)
+
+        if not isinstance(maybe_fast_module, bool):
+            fast_module = maybe_fast_module(module)
+            model._modules[name] = fast_module
+        elif hasattr(module, "_hf_hook"):
+            # If the module has `accelerate` hooks, manually
+            # set the parameters on the `CPU` if there is
+            # any offload involved.
+            module = init_accelerate_hook(module)
+            model._modules[name] = module
+    return model
+
+
+# Step 5: Post process the potentially converted model by setting the `is_last_layer` attribute to `True`
+# For the last `Fast` layer.
+def set_last_layer(model):
+    r"""
+    Iterates over the module list containing the `LayerFast` modules. Sets the last layer's `is_last_layer`
+    attribute to `True`
+
+    Args:
+        `model` (`torch.nn.Module`, **required**):
+            The input converted model
+    Returns:
+        Returns `True` if it has succesfully set the attribute to `True`, otherwise return `False`.
+    """
+    dict_named_module = dict(model.named_modules())
+    sort_fn = lambda list_modules: [module.__class__.__name__ for module in list_modules]  # noqa: E731
+
+    for key in dict_named_module.keys():
+        if isinstance(dict_named_module[key], torch.nn.ModuleList) and all(
+            "LayerFast" in module_name for module_name in sort_fn(dict_named_module[key])
+        ):
+            setattr(dict_named_module[key][-1], "is_last_layer", True)
+            return True
+    return False
+
+
+@check_if_pytorch_greater_112()
+class BetterTransformer(object):
+    r"""
+    A conversion wrapper that takes as an input the `transformers` model to be converted
+    and returns the converted `Fast` model. The `Fast` model is based on the `BetterTransformer`
+    recently released by PyTorch from its 1.12 version:
+    https://pytorch.org/blog/a-better-transformer-for-fast-transformer-encoder-inference/
+
+    Original PR from: https://github.com/huggingface/transformers/pull/19553 adapted and wrapped in
+    this script.
+    """
+
+    def transform(model, keep_original_model=False, max_memory=None):
```

Comment:
maybe add type hints + return hint

Reply:
Thanks! Fixed in 3d19a755c390c2444789055f2fc9adb1affbfb07

Reply:
.. and in 633fa43 !